src/einsteinpy/geodesic/geodesic.py

```diff
@@ -100,60 +144,269 @@ def trajectory(self):
+            Defaults to ``False``
 
-        return lambdas, vecs
+        """
+        super().__init__(
+            metric=metric,
+            metric_params=metric_params,
+            position=position,
+            momentum=momentum,
+            time_like=False,
+            return_cartesian=return_cartesian,
+            **kwargs,
+        )
+
+
+class Timelike(Geodesic):
+    """
+    Class for defining Time-like Geodesics
+
+    """
+
+    def __init__(
+        self, metric, metric_params, position, momentum, return_cartesian=True, **kwargs
+    ):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        metric : str
+            Name of the metric. Currently, these metrics are supported:
+            1. Schwarzschild
+            2. Kerr
+            3. KerrNewman
+        metric_params : array_like
+            Tuple of parameters to pass to the metric
+            E.g., ``(a,)`` for Kerr
+        position : array_like
+            3-Position
+            4-Position is initialized by taking ``t = 0.0``
+        momentum : array_like
+            3-Momentum
+            4-Momentum is calculated automatically,
+            considering the value of ``time_like``
+        return_cartesian : bool, optional
+            Whether to return calculated positions in Cartesian Coordinates
+            This only affects the coordinates. The momenta dimensionless
+            quantities, and are returned in Spherical Polar Coordinates.
+            Defaults to ``True``
+        kwargs : dict
+            Keyword parameters for the Geodesic Integrator
+            See 'Other Parameters' below.
+
+        Other Parameters
+        ----------------
+        steps : int
+            Number of integration steps
+            Defaults to ``50``
+        delta : float
+            Initial integration step-size
+            Defaults to ``0.5``
+        rtol : float
+            Relative Tolerance
+            Defaults to ``1e-2``
+        atol : float
+            Absolute Tolerance
+            Defaults to ``1e-2``
+        order : int
+            Integration Order
+            Defaults to ``2``
+        omega : float
+            Coupling between Hamiltonian Flows
+            Smaller values imply smaller integration error, but too
+            small values can make the equation of motion non-integrable.
+            For non-capture trajectories, ``omega = 1.0`` is recommended.
+            For trajectories, that either lead to a capture or a grazing
+            geodesic, a decreased value of ``0.01`` or less is recommended.
+            Defaults to ``1.0``
+        suppress_warnings : bool
+            Whether to suppress warnings during simulation
+            Warnings are shown for every step, where numerical errors
+            exceed specified tolerance (controlled by ``rtol`` and ``atol``)
+            Defaults to ``False``
+
+        """
+        super().__init__(
```

Comment:
But since it's a very thin wrapper around Geodesics, do we really want the separation?

Reply:
It's more of a convenience thing, really. I agree, it is a thin wrapper, but for the time being, I'd keep it, as the Radiative Transfer module might also need some additions (extra methods) to `Nulllike`.